### PR TITLE
106 distinguish between iota nodes

### PR DESCRIPTION
**Pull request by dominic22**
- update iota sdk to 0.3.4
- use permanode to resolve identities
- refactor streamsstate
- adjust env variables and example

### CODE REVIEW

**Comment by dominic22** [comment]
@dominic22 adjust repository secret for hornet and perma node

File: api/src/services/ssi-service.ts [hunk 1]
```diff
@@ -138,7 +140,8 @@ export class SsiService {
 	}
 
 	async publishSignedDoc(newDoc: IdentityDocumentJson): Promise<string> {
-		const txHash = await Identity.publish(newDoc, this.config);
+		const client = this.getIdentityClient();
+		const txHash = await client.publishDocument(newDoc);
```
Comment (sobolev-alexey) [inline api/src/services/ssi-service.ts:144]: looks like the new version introduced some breaking changes

File: api/src/services/ssi-service.ts [hunk 2]
```diff
@@ -209,7 +213,7 @@ export class SsiService {
 
 	generateIdentity() {
 		try {
-			const { doc, key } = new Document(this.config.keyType) as IdentityDocument;
+			const { doc, key } = new Document(this.config.keyType, Identity.Network.mainnet().toString()) as IdentityDocument;
```
Comment (sobolev-alexey) [inline api/src/services/ssi-service.ts:216]: do they now hardcode the Node URL inside the lib?

**Review by sobolev-alexey — APPROVED** [review]

File: api/src/services/ssi-service.ts (hunk 1 above)
Comment (dominic22) [reply api/src/services/ssi-service.ts:144]: yes there were some breaking changes but easily to handle

File: api/src/services/ssi-service.ts (hunk 2 above)
Comment (dominic22) [reply api/src/services/ssi-service.ts:216]: Oh yes they have pointing to the iota nodes, but I've wanted to replace them by the defined env var. I will adjust it.

**Pull request merged**